modules/auxiliary/scanner/rdp/cve_2019_0708_bluekeep.rb

```diff
@@ -564,6 +580,489 @@ def int_to_bytestring( daInt, num_chars=nil )
+    "\x78\x00\x00\x00\x50\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" +
+    "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" +
+    "\x08\x00" + # capabilitySetType: 8 - TS_POINTER_CAPABILITYSET
+    "\x0a\x00" + # lengthCapability: 10
+    "\x01\x00\x14\x00\x14\x00" +
+    "\x0a\x00" + # capabilitySetType: 10 - TS_COLORTABLE_CAPABILITYSET
+    "\x08\x00" + # lengthCapability: 8
+    "\x06\x00\x00\x00" +
+    "\x07\x00" + # capabilitySetType: 7 - TSWINDOWACTIVATION_CAPABILITYSET
+    "\x0c\x00" + # lengthCapability: 12
+    "\x00\x00\x00\x00\x00\x00\x00\x00" +
+    "\x05\x00" + # capabilitySetType: 5 - TS_CONTROL_CAPABILITYSET
+    "\x0c\x00" + # lengthCapability: 12
+    "\x00\x00\x00\x00\x02\x00\x02\x00" +
+    "\x09\x00" + # capabilitySetType: 9 - TS_SHARE_CAPABILITYSET
+    "\x08\x00" + # lengthCapability: 8
+    "\x00\x00\x00\x00" +
+    "\x0f\x00" + # capabilitySetType: 15 - TS_BRUSH_CAPABILITYSET
+    "\x08\x00" + # lengthCapability: 8
+    "\x01\x00\x00\x00" +
+    "\x0d\x00" + # capabilitySetType: 13 - TS_INPUT_CAPABILITYSET
+    "\x58\x00" + # lengthCapability: 88
+    "\x01\x00\x00\x00\x09\x04\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00" +
+    "\x0c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" +
+    "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" +
+    "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" +
+    "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" +
+    "\x00\x00\x00\x00" +
+    "\x0c\x00" + # capabilitySetType: 12 - TS_SOUND_CAPABILITYSET
+    "\x08\x00" + # lengthCapability: 8
+    "\x01\x00\x00\x00" +
+    "\x0e\x00" + # capabilitySetType: 14 - TS_FONT_CAPABILITYSET
+    "\x08\x00" + # lengthCapability: 8
+    "\x01\x00\x00\x00" +
+    "\x10\x00" + # capabilitySetType: 16 - TS_GLYPHCAChE_CAPABILITYSET
+    "\x34\x00" + # lengthCapability: 52
+    "\xfe\x00\x04\x00\xfe\x00\x04\x00\xfe\x00\x08\x00\xfe\x00\x08\x00" +
+    "\xfe\x00\x10\x00\xfe\x00\x20\x00\xfe\x00\x40\x00\xfe\x00\x80\x00" +
+    "\xfe\x00\x00\x01\x40\x00\x00\x08\x00\x01\x00\x01\x02\x00\x00\x00"
+
+    # type = 0x13 = TS_PROTOCOL_VERSION | PDUTYPE_CONFIRMACTIVEPDU
+    build_share_control_header(0x13, pdu)
+  end
+
+  # https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/ff7f06f8-0dcf-4c8d-be1f-596ae60c4396
+  # Client Input Event Data - TS_INPUT_PDU_DATA - 2.2.8.1.1.3.1
+  def pdu_client_input_event_sychronize
+
+    pdu = "\x01\x00" +   # numEvents: 1
+    "\x00\x00" +         # pad2Octets
+    "\x00\x00\x00\x00" + # eventTime
+    "\x00\x00" +         # messageType: 0 - INPUT_EVENT_SYNC
+    # TS_SYNC_EVENT 202.8.1.1.3.1.1.5
+    "\x00\x00" +         # pad2Octets
+    "\x00\x00\x00\x00" + # toggleFlags
+    # The data below is not part of the specification
+    # Testing without the data did not discover any impact on reliability
+    # or target stablility. Will likely be removed in a future PR.
```

Comment:
@zerosum0x0 - The payload is a Client Input Event (TS_INPUT_PDU - 2.2.8.1.1.3) with messageType Synchronize Event (TS_SYNC_EVENT - 2.2.8.1.1.3.1.5) which shouldn't have (keyboard/mouse) input data.  I don't see any reason (from a protocol perspective) for the the 800 or so a bytes. Does this influence the stability or patch detection?

Reply:
I don't think it matters. It's just a paste from what rdesktop was saying to the server. Probably has fat that can be trimmed, would need to dig into the message syntax.

Reply:
Thanks for the response.. I can't reproduce it and It doesn't seem to impact the stability of the check. I've trimmed the data out.